script.recap/resources/lib/core.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+import xbmc
+import xbmcaddon
+import xbmcgui
+
+# Import the common settings
+from settings import log
+from settings import Settings
+from recap import Recap
+from viewer import Viewer
+
+ADDON = xbmcaddon.Addon(id='script.recap')
+
+
+#########################
+# Main
+#########################
+def runRecap(previous=False):
+    log("runRecap: Started")
+
+    # First check to see if we have a TV Show already selected
+    showName = None
+    if xbmc.getInfoLabel("ListItem.dbtype") in ['tvshow', 'season', 'episode']:
```

Comment:
Just a small note on sets (you got me looking them up as I have never used them) the {} notation apparently only works in python 2.7+ which I don't think we can't count on until Krypton. Ignorable opinion follows: In a three element/single iteration like this I would personally use a tuple - based on the assumption that it is a lower resource using object and also that in most cases (I believe) with very small sets of data you don't get any benefit from more complex lookup algorithms.